### DISCUSSION TITLE
counting the numbers of records ?

### DISCUSSION THREAD

**Discussion by Matrex7** [pocketbase/pocketbase#3513] [Q&A] [answered]
so you can count the number of records in sql with count(record) but idk how in this , like a post table that have comments ; how to get numbers of comments, i usually  add a number records associated with the comments as commentsCount for example , how do i achieve this in pocketbase .

**Comment by ganigeorgiev** [comment]
It is not clear from your description what count exactly you want, but in general for count you can send a `getList` request with `perPage=1` OR create helper view collection with custom SQL if you need more advanced aggregations.

```js
const result = await pb.collection("example").getList(1, 1)

console.log(result.totalItems);
```

**Reply by ganigeorgiev** [reply, marked as answer]
On second read, if you want to count the records based on some `relation` field, you can create View collection with a query like:

```sql
SELECT
    comments.post as id,
    comments.post, -- optional field to allow expand of the post record
    count(comments.id) as totalComments
FROM comments
GROUP BY comments.post
```
_("post" in the above is a `relation` field in the "comments" collection; the above should create a new collection with 3 fields - "id", "post" and "totalComments")_
